src/EnergyEvaluation.jl

```diff
@@ -375,6 +377,135 @@ function dimer_energy(pot::ELJPotentialEven{N}, r2) where N
     end
     return sum1
 end 
+#-----------------------------------------------------------------------------#
+#----------------------------EmbeddedAtomPotential----------------------------#
+#-----------------------------------------------------------------------------#
+"""
+    EmbeddedAtomPotential
+Struct containing the important quantities for calculating EAM (specifically Sutton-Chen type) potentials.
+    Fields:
+    `n` the exponent for the two-body repulsive ϕ component
+    `m` the exponent for the embedded electron density ρ
+    `ean` multiplicative factor ϵa^n /2 for ϕ
+    `eCam` multiplicative factor ϵCa^(m/2) for ρ 
+
+"""
+struct EmbeddedAtomPotential <: AbstractDimerPotential
+    n::Float64
+    m::Float64
+    ean::Float64
+    eCam::Float64
+end
+"""
+    EmbeddedAtomPotential(n,m,ϵ,C,a)
+Function to initalise the EAM struct given the actual constants cited in papers. The exponents `n`,`m`, the energy constant `ϵ` the distance constant `a` standard in all EAM models, and a dimensionless parameter `C` scaling ρ with respect to ϕ.
+"""
+function EmbeddedAtomPotential(n,m,ϵ,C,a)
+    epsan = ϵ * a^n / 2
+    epsCam = ϵ * C * a^(m/2)
+    return EmbeddedAtomPotential(n,m,epsan,epsCam)
+end
+"""
+    invrexp(r2,n,m)
+Function to calculate the `ϕ,ρ` components given a square distance `r2` and the exponents `n,m`
+"""
+function invrexp(r2,n,m)
+    if r2 != 0.
+        r_term = 1/sqrt(r2)
+        return r_term^n , r_term^m
+    else
+        return 0. , 0.
+    end    
+end
+"""
+    calc_components(eatomvec,distancevec,n,m)
+Primary calculation of ϕ,ρ for atom i, given each other atom's distance to i in `distancevec`. `eatomvec` stores the ϕ and ρ components.
+"""
+function calc_components(eatomvec,distancevec,n,m)
+    for dist in distancevec
+        eatomvec .+= invrexp(dist,n,m)
+    end
+    return eatomvec
+end
+"""
+    calc_all_components(dist2_mat,n,m) 
+    calc_components(componentvec,atomindex,old_r2_vec,new_r2_vec,n,m)
+Given a matrix `dist2_mat` of each square-radial distance and the exponents `n,m` calculates the Nx2 vector of ϕ and ρ components for each atom and returning them as a `component_vector`.
+
+Second method also includes an existing `componentvec` `atomindex` and old and new interatomic distances from an atom at `atomindex` stored in vectors `new_r2_vec,old_r2_vec`. This calculates the `new_component_vec` based on the updated distances and returns this.
+"""
```

Comment:
I can't find any function calc_all_components?
All calc_components methods should be described in one docu string.

Reply:
This is a leftover from an earlier version where the two methods were separate functions, now corrected